integration-tests/pkg/mock_sensor/server.go

```diff
@@ -432,11 +432,10 @@ func (m *MockSensor) pushConnection(containerID string, connection *sensorAPI.Ne
 		CloseTimestamp: connection.GetCloseTimestamp().String(),
 	}
 
-	if connections, ok := m.connections[containerID]; ok {
-		connections[conn] = true
+	if _, ok := m.connections[containerID]; ok {
+		m.connections[containerID] = append(m.connections[containerID], conn)
```

Comment:
[nit] Not 100% sure this will work.
```suggestion
	if c, ok := m.connections[containerID]; ok {
		m.connections[containerID] = append(c, conn)
```